packages/muelu/test/unit_tests/RegionVector.cpp

```diff
@@ -0,0 +1,381 @@
+  lNodesPerDir[0] = galeriList.get<LO>("lnx");
+  lNodesPerDir[1] = galeriList.get<LO>("lny");
+  lNodesPerDir[2] = 1;
+  procsPerDim[0] = galeriList.get<GO>("mx");
+  procsPerDim[1] = galeriList.get<GO>("my");
+  procsPerDim[2] = 1;
+
+  Array<int> boundaryConditions;
+  int maxRegPerGID = 0;
+  int numInterfaces = 0;
+  LO numLocalRegionNodes = 0;
+  Array<GO>  sendGIDs;
+  Array<int> sendPIDs;
+  Array<LO>  rNodesPerDim(3);
+  Array<LO>  compositeToRegionLIDs(nodeMap->getNodeNumElements()*numDofsPerNode);
+  Array<GO>  quasiRegionGIDs;
+  Array<GO>  quasiRegionCoordGIDs;
+  createRegionData(2, false, numDofsPerNode,
+                   gNodesPerDir(), lNodesPerDir(), procsPerDim(),
+                   nodeMap, dofMap,
+                   maxRegPerGID, numLocalRegionNodes, boundaryConditions,
+                   sendGIDs, sendPIDs, numInterfaces, rNodesPerDim,
+                   quasiRegionGIDs, quasiRegionCoordGIDs, compositeToRegionLIDs);
+
+  const int maxRegPerProc = 1;
+  std::vector<RCP<Map> > rowMapPerGrp(maxRegPerProc),        colMapPerGrp(maxRegPerProc);
+  rowMapPerGrp[0] = Xpetra::MapFactory<LO,GO,Node>::Build(A->getRowMap()->lib(),
+                                                          Teuchos::OrdinalTraits<GO>::invalid(),
+                                                          quasiRegionGIDs(),
+                                                          A->getRowMap()->getIndexBase(),
+                                                          A->getRowMap()->getComm());
+  colMapPerGrp[0] = rowMapPerGrp[0];
+
+  std::vector<RCP<Map> > revisedRowMapPerGrp(maxRegPerProc), revisedColMapPerGrp(maxRegPerProc);
+  revisedRowMapPerGrp[0] = Xpetra::MapFactory<LO,GO,Node>::Build(A->getRowMap()->lib(),
+                                                                 Teuchos::OrdinalTraits<GO>::invalid(),
+                                                                 quasiRegionGIDs.size()*numDofsPerNode,
+                                                                 A->getRowMap()->getIndexBase(),
+                                                                 A->getRowMap()->getComm());
+  revisedColMapPerGrp[0] = revisedRowMapPerGrp[0];
+
+  std::vector<RCP<Import> > rowImportPerGrp(maxRegPerProc);
+  std::vector<RCP<Import> > colImportPerGrp(maxRegPerProc);
+  rowImportPerGrp[0] = ImportFactory::Build(dofMap, rowMapPerGrp[0]);
+  colImportPerGrp[0] = ImportFactory::Build(dofMap, colMapPerGrp[0]);
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  { // test compositeToRegional
+    // Create a composite vector
+    RCP<Vector> compVec = VectorFactory::Build(dofMap);
+    TEUCHOS_ASSERT(!compVec.is_null());
+    compVec->putScalar(0.0);
+    const size_t localLength = compVec->getLocalLength();
+    for (size_t k = 0; k < localLength; ++k) {
+      compVec->replaceLocalValue(k, k + myRank/10.0);// (LO, Scalar) need to fix type of k
```

Comment:
To fix type of k you only need to use `static_cast<some_new_type>(myOldTypeVar)`

Reply:
@pohm Do you have time to take care of this? If not, please let me know. Then, I will look at it in detail.